pandas/tests/indexing/test_loc.py

```diff
@@ -2117,6 +2134,14 @@ def test_loc_setitem_dt64tz_values(self):
         result = s2["a"]
         assert result == expected
 
+    @pytest.mark.parametrize("dtype", ["int64", "Int64"])
+    def test_loc_setitem_series_null_slice_different_dtypes(self, dtype):
+        # GH#20635
+        ser = Series(["3", "4"], name="A")
+        ser.loc[:] = ser.astype("int64").astype(dtype)
+        expected = Series([3, 4], name="A", dtype=dtype)
```

Comment:
i think this is doing the opposite of #39163.  did we decide to revert part or all of that?

Reply:
Since yours is significantly newer I am fine with „closing“ this. Would check if some of the issues are fixed